---
description: Advertencia del compilador (nivel 1) CS1058
title: Advertencia del compilador (nivel 1) CS1058
ms.date: 07/20/2015
f1_keywords:
- CS1058
helpviewer_keywords:
- CS1058
ms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d
ms.openlocfilehash: a03077a7b7671f0affd003b2d281aea902c9af61
ms.sourcegitcommit: 5b475c1855b32cf78d2d1bbb4295e4c236f39464
ms.translationtype: HT
ms.contentlocale: es-ES
ms.lasthandoff: 09/24/2020
ms.locfileid: "91179300"
---
# <a name="compiler-warning-level-1-cs1058"></a><span data-ttu-id="1cf0f-103">Advertencia del compilador (nivel 1) CS1058</span><span class="sxs-lookup"><span data-stu-id="1cf0f-103">Compiler Warning (level 1) CS1058</span></span>

<span data-ttu-id="1cf0f-104">Una cláusula catch previa ya detecta todas las excepciones.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-104">A previous catch clause already catches all exceptions.</span></span> <span data-ttu-id="1cf0f-105">Todas las excepciones producidas se ajustarán en System.Runtime.CompilerServices.RuntimeWrappedException</span><span class="sxs-lookup"><span data-stu-id="1cf0f-105">All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException</span></span>  
  
 <span data-ttu-id="1cf0f-106">Este atributo provoca una advertencia CS1058 si un bloque `catch()` no tiene ningún tipo de excepción especificado después de un bloque `catch (System.Exception e)`.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-106">This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block.</span></span> <span data-ttu-id="1cf0f-107">La advertencia le avisa de que el bloque `catch()` no capturará ninguna excepción.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-107">The warning advises that the `catch()` block will not catch any exceptions.</span></span>  
  
<span data-ttu-id="1cf0f-108">Un bloque `catch()` después de un bloque `catch (System.Exception e)` puede capturar excepciones de no compatibilidad con CLS si `RuntimeCompatibilityAttribute` se establece en False en el archivo AssemblyInfo.cs: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-108">A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`.</span></span> <span data-ttu-id="1cf0f-109">Si este atributo no se establece explícitamente en False, todas las excepciones de no compatibilidad con CLS generadas se ajustan como excepciones y el bloque `catch (System.Exception e)` las detecta.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-109">If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them.</span></span> <span data-ttu-id="1cf0f-110">Para más información, consulte [Procedimiento para detectar excepciones no compatibles con CLS](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span><span class="sxs-lookup"><span data-stu-id="1cf0f-110">For more information, see [How to catch a non-CLS exception](../../programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).</span></span>
  
## <a name="example"></a><span data-ttu-id="1cf0f-111">Ejemplo</span><span class="sxs-lookup"><span data-stu-id="1cf0f-111">Example</span></span>  

 <span data-ttu-id="1cf0f-112">En el ejemplo siguiente se genera el error CS1058.</span><span class="sxs-lookup"><span data-stu-id="1cf0f-112">The following example generates CS1058.</span></span>  
  
```csharp  
// CS1058.cs  
// CS1058 expected  
using System.Runtime.CompilerServices;  
  
// the following attribute is set to true by default in the C# compiler  
// set to false in your source code to resolve CS1058  
[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  
  
class TestClass
{  
   static void Main()
   {  
      try {}  
  
      catch (System.Exception e) {
         System. Console.WriteLine("Caught exception {0}", e);  
      }  
  
      catch {}   // CS1058. This line will never be reached.  
   }  
}  
```
